lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -98,3 +106,153 @@ def _validate_and_link(program_uuid_string, linkage_text):
+        if edx_username_or_email:
+            learner_program_enrollments, error = self._get_account_info(edx_username_or_email)
+            if error:
+                errors.append(error)
+        elif org_key and external_user_key:
+            learner_program_enrollments = {}
+        elif not external_user_key and org_key:
+            errors.append(
+                'You must provide either the edX username or email, or the '
+                'Learner Account Provider and External Key pair to do search!'
+            )
+
+        return render_to_response(
+            self.CONSOLE_TEMPLATE_PATH,
+            {
+                'successes': [],
+                'errors': errors,
+                'learner_program_enrollments': learner_program_enrollments,
+                'org_keys': self._get_org_keys_with_IdP_provider(),
+            }
+        )
+
+    def _get_org_keys_with_IdP_provider(self):
+        """
+        From our Third_party_auth models, return a list
+        of organizations whose SAMLProviders are active and configured
+        """
+        saml_providers = SAMLProviderConfig.objects.current_set().filter(
+            enabled=True,
+            organization__isnull=False
+        ).select_related('organization')
+
+        return [saml_provider.organization.short_name for saml_provider in saml_providers]
+
+    def _get_account_info(self, username_or_email):
+        """
+        Provided the edx account username or email, return edx account info
+        and program_enrollments_info. If we cannot identify the user, return
+        empty object and error.
+        """
+        user_info = {}
+        external_key = None
+        try:
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            user_info['username'] = user.username
+            user_info['email'] = user.email
+            try:
+                user_social_auth = UserSocialAuth.objects.get(user=user)
+                _, external_key = user_social_auth.uid.split(':', 1)
+                user_info['external_user_key'] = external_key
+                user_info['SSO'] = {
+                    'uid': user_social_auth.uid,
+                    'provider': user_social_auth.provider
+                }
+            except UserSocialAuth.DoesNotExist:
+                pass
+
+            enrollments = self._get_enrollments(user=user)
+            result = {'user': user_info}
+            if enrollments:
+                result['enrollments'] = enrollments
+
+            return result, ''
+        except User.DoesNotExist:
+            return {}, 'Could not find edx account with {}'.format(username_or_email)
+
+    def _get_enrollments(self, user=None, external_user_key=None):
+        """
+        With the user or external_user_key passed in,
+        return a dictionary with corresponding ProgramEnrollments and ProgramCourseEnrollments
+        all serialized for view
+        """
```

Comment:
We're returning a list of dictionaries here, not a dictionary

Reply:
Feedback taken